language/go/generate.go

```diff
@@ -112,6 +122,11 @@ func (gl *goLang) GenerateRules(args language.GenerateArgs) language.GenerateRes
 						importPath: goProtoImportPath(gc, ppkg, args.Rel),
 						proto:      protoTargetFromProtoPackage(name, ppkg),
 					}
+					if r, ok := goProtoRules[":"+name]; ok {
+						// if a go_proto_library rule already exists for this
```

Comment:
This means if the directory:
  1. `# gazelle:proto default` or `"-proto=default"`
  2. has no go files, and
  3. only one proto package,
the `go_library` rule generated will use the `go_proto_library` rule's import path.

I'm not sure if this is the ideal behavior, and I'm marking this one up for discussion. Let me know if you have any other thoughts/ideas. @jayconrod

Reply:
Integrating `go_library` with `go_proto_library` generated by another extension seems too hard. There are just too many corner cases and too many things that could go wrong.
 
If a `proto_library` is covered by a `go_proto_library` generated by another extension, I think we should simply proceed as if neither rule exists. The Go extension should not generate a `go_proto_library` for that proto, and it shouldn't generate a `go_library` if there aren't any other .go files in the directory.

Reply:
SGTM.

I assume this means that if we have colliding import paths with a `go_proto_library` and a `go_library`, we'll just have separate rules with the same import path, rather than embedding the `go_proto_library` rule in the `go_library` rule, correct?